File: NickYMartinApi/NickYMartinApi/Models/Carrito.cs
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace NickYMartinApi.Models
{
    public class Carrito
    {
        [Key]
        public Guid IdCarrito { get; set; }
        [ForeignKey(nameof(Usuario))]
        public Guid? IdUsuario { get; set; }
        public virtual User? Usuario { get; set; }
        public string? userMail { get; set; }
        public DateTime FechaCreacion { get; set; }
        public virtual ICollection<ItemCarrito>? ItemsCarrito { get; set; }
        public decimal Total { get; set; }
    }
}

File: NickYMartinApp/src/types/carrito.ts
import type { Producto } from './product';

export interface Carrito {
    idCarrito: string; 
    idUsuario: string; 
    fechaCreacion: string; 
    fechaActualizacion: string; 
    itemsCarrito?: ItemCarrito[]; 
    total: number;
}

export interface ItemCarrito {
    idItemCarrito: string; 
    idCarrito: string;     
    carrito?: Carrito;    

    idProducto: string;
    producto?: Producto;  

    mainImageUrl?: string;
    cantidad: number;      
    precioUnitario: number; 
    subtotal: number;       

    fechaCreacion: string; 
}